app/repositories/proxy_types_repository.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+class ProxyTypesRepository
+  attr_reader :base_url, :headers
+
+  def initialize(config:)
+    @config = config
+    @base_url = "#{rmapi_url}/rm/rmaccounts/#{config.customer_id}/proxies"
+
+    @headers = {
+      'X-Api-Key': config.api_key,
+      'Content-Type': 'application/json',
+      'Accept': 'application/json'
+    }
+  end
+
+  def where!
+    request do
+      status, body = rmapi(:get, '')
+      Result.new(
+        data: body.map { |hash| to_proxy_type hash },
+        status: status
+      )
+    end
+  end
+
+  private
+
+  # superclass for repository errors
+  class RepositoryError < StandardError; end
```

Comment:
these also would be easy - maybe even easier than the stuff in `initialize`.  they're class definitions, we can just put them in a separate file (or files) in the same namespace and they should be usable by all repos.  no inheritance or anything, just literal copy-paste